addons/web/static/tests/views/form_tests.js

```diff
@@ -5557,6 +5557,44 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test('open new record even with warning message', async function (assert) {
+        assert.expect(3);
+
+        this.data.partner.onchanges = { foo: true };
+
+        var form = await createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: '<form string="Partners">' +
+                    '<group><field name="foo"/></group>' +
+                '</form>',
+            res_id: 2,
+            mockRPC: function (route, args) {
+                if (args.method === 'onchange') {
+                    return Promise.resolve({
+                        warning: {
+                            title: "Warning",
+                            message: "Any warning."
+                        }
+                    });
+                }
+                return this._super.apply(this, arguments);
+            },
+
+        });
+        await testUtils.dom.click(form.$buttons.find('.o_form_button_edit'));
+        assert.strictEqual(form.$('input').val(), 'blip', 'input should contain record value');
+        form.$('input').first().val("tralala").trigger('input');
+        assert.strictEqual(form.$('input').val(), 'tralala', 'input should contain new value');
+
+        await form.reload({ currentId: false });
```

Comment:
@aab-odoo @nle-odoo 
Is it normal that in 14.0, after a reload, it does not take the default value of the field ('My little Foo Value')?

Reply:
@nboulif The reason is that as of 14.0, default_get and onchange requests have been merged. Before v14, we did a default_get (which returned the default value) and then an onchange (which returned the warning, as stated in mockRPC). Since now there is a single call (which theoretically returns the default values AND the onchanges), we only get the result of the onchange as stated in mockRPC. So to summarize, yes it is normal, according how the test is written.

Reply:
Thanks for the explanation! :+1: